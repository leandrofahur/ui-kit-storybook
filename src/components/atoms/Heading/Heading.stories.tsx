import React from "react";
import { Story, Meta } from "@storybook/react";
import Heading, { HeadingProps } from "./Heading";
import styled from "styled-components";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
    title: "Components/atoms/Heading",
    component: Heading,
    argTypes: {},
} as Meta<HeadingProps>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: Story<HeadingProps> = (args) => <Heading {...args} />;

const Container = styled.div`
    display: flex;
    flex-direction: column;

    & > * {
        padding-top: 10px;
    }
`;

export const All = () => {
    return (
        <Container>
            <h5>Typography</h5>
            <Heading as="h6">Heading h6</Heading>
            <Heading as="h5">Heading h5</Heading>
            <Heading as="h4">Heading h4</Heading>
            <Heading as="h3">Heading h3</Heading>
            <Heading as="h2">Heading h2</Heading>
            <Heading as="h1">Heading h1</Heading>
        </Container>
    );
};
