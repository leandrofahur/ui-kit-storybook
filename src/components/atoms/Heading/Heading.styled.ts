import styled from "styled-components";

export const Title = styled.div`
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        padding-bottom: 20px;
    }
`;
