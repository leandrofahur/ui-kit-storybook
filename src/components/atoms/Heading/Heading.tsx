import React, { ElementType } from "react";
import { Title } from "./Heading.styled";

export interface HeadingProps {
    /** Props to specify the element as h1, h2, h3 ... h6 */
    as: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
    /** Provides the inner text for the tag */
    children: string;
}

const Heading = ({ as, children }: HeadingProps) => {
    return <Title as={as}>{children}</Title>;
};

export default Heading;
