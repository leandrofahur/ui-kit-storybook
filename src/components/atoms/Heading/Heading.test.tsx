import Heading from "./Heading";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

test("Renders Heading with children and matches with snapshot", () => {
    const { asFragment } = render(<Heading as="h1">Hello World</Heading>);
    expect(asFragment()).toMatchSnapshot();
});
