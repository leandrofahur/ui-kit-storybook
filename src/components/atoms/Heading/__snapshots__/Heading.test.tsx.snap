// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Heading with children and matches with snapshot 1`] = `
<DocumentFragment>
  .c0 h1,
.c0 h2,
.c0 h3,
.c0 h4,
.c0 h5,
.c0 h6 {
  padding-bottom: 20px;
}

<h1
    class="c0"
  >
    Hello World
  </h1>
</DocumentFragment>
`;
